### PR TITLE
CLN: Simplify aggregation.aggregate

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/aggregation.py [hunk 1]
```diff
@@ -608,92 +608,43 @@ def aggregate(obj, arg: AggFuncType, *args, **kwargs):
...
-            result = {}
+        results = {}
+        if selected_obj.ndim == 1:
+            # fname only used for output
+            colg = obj._gotitem(obj._selection, ndim=1)
             for fname, agg_how in arg.items():
-                result[fname] = func(fname, agg_how)
-            return result
+                results[fname] = colg.aggregate(agg_how)
+        else:
```
Comment (jreback) [inline pandas/core/aggregation.py:617]: can you make these dict-comprehensions?

**Review by jreback — CHANGES_REQUESTED** [review]
wow

**Review by jreback — APPROVED** [review]
lgtm merge on green

File: pandas/core/aggregation.py [hunk 2]
```diff
@@ -608,92 +608,41 @@ def aggregate(obj, arg: AggFuncType, *args, **kwargs):
 
         from pandas.core.reshape.concat import concat
 
-        def _agg_1dim(name, how, subset=None):
-            """
-            aggregate a 1-dim with how
-            """
-            colg = obj._gotitem(name, ndim=1, subset=subset)
-            if colg.ndim != 1:
-                raise SpecificationError(
```
Comment (jbrockmendel) [inline pandas/core/aggregation.py:617]: i take it this is never reachable?  IIUC thats the lynchpin to the cleanup

Comment (rhshadrach) [reply pandas/core/aggregation.py:617]: That's right; this is only ever called via `_agg`, so `name` here is a key in `arg`. It must therefore be a single label.

**Comment by jreback** [comment]
nice, thanks @rhshadrach keep em coming

**Pull request merged**